Services/ReleaseNoteService.cs

```diff
@@ -23,9 +27,41 @@ public ReleaseNoteService(IReleaseNoteRepository releaseNoteRepository, IUnitOfW
             _mapper = mapper;
         }
 
-        public async Task<IEnumerable<ReleaseNote>> ListAsync()
+        public async Task<IEnumerable<ReleaseNote>> ListAsync(ReleaseNoteParameters queryParameters)
         {
-            return await _releaseNoteRepository.ListAsync();
+            return await _releaseNoteRepository.ListAsync(queryParameters);
+        }
+
+        public async Task<ReleaseNotesResponse> FilterDates(IEnumerable<ReleaseNote> notes,
+            ReleaseNoteParameters queryParameters)
+        {
+            try
+            {
+                DateTime start = queryParameters.StartDate.Value;
+                DateTime end = queryParameters.EndDate.Value;
+                int res = start.CompareTo(end);
+                if (res < 0)
+                {
+                    // Ordinary filter request
+                    var validNotes = notes.Where(rn => rn.ClosedDate >= start && rn.ClosedDate <= end);
+                    return new ReleaseNotesResponse(validNotes.ToList());
+                }
+
+                if (res == 0)
+                {
+                    // StartDate and EndDate are the same
+                    var validNotes = notes.Where(rn =>
+                        rn.ClosedDate.CompareTo(start) == 0);
+                    return new ReleaseNotesResponse(validNotes.ToList());
+                }
+            }
+            catch (Exception e)
+            {
+                // if none of the above is entered => assume that the query is incorrect:
+                return new ReleaseNotesResponse($"Det oppsto en feil: {e.Message}");
+            }
+
+            return new ReleaseNotesResponse("Det oppsto en feil!");
         }
```

Comment:
Det her vil eg sei e en jobb for databasen, ihvertfall ikke nå vits i å hent ut alle Release Notes, førr så å sorter ut de du ikke treng.

Vess du heller lag en query som spørr etter akkurat de Release Notesan du ska ha, så bi det ei mye bedre løsning.

Reply:
de nyeste commitsene gjør dette nå